Source/SwiftLintFramework/Rules/FileprivateRule.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  FileprivateRule.swift
+//  SwiftLint
+//
+//  Created by Jose Cheyo Jimenez on 05/02/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct FileprivateRule: Rule, ConfigurationProviderRule {
+    public var configuration = FileprivateConfiguration(strict: false)
+
+    public init() {}
+
+    public static let description = FileprivateConfiguration.fileprivateLimited
+
+    public func validate(file: File) -> [StyleViolation] {
+        if !configuration.strict {
+            let toplevel = file.structure.dictionary.substructure.flatMap({ $0.offset })
+            let syntaxTokens = file.syntaxMap.tokens
+            let violationOffsets = toplevel.flatMap { (offSet) -> Int? in
+                let parts = syntaxTokens.partitioned { offSet <= $0.offset }
+                guard let lastKind = parts.first.last,
+                    lastKind.length == "fileprivate".bridge().length,
```

Comment:
Full disclosure here: There is another builtin that has the same length as `fileprivate`. `nonmutating` has the same length of characters but `nonmutating` can never be toplevel in the current version of swift AFAIK. We could just `file.substring` to find out exactly what the builtin key is but that would slow down this pass.